### PR TITLE
Add half float mapping to the scripting fields API

### PR DESCRIPTION
**Pull request by jdconrad**
This adds the mapped type half float the scripting fields API. This also adds additional methods for `asDouble` to reach parity with old-style `doc` access.

### CODE REVIEW

File: modules/lang-painless/src/main/resources/org/elasticsearch/painless/org.elasticsearch.script.fields.txt [hunk 1]
```diff
@@ -61,6 +61,14 @@ class org.elasticsearch.script.field.ScaledFloatDocValuesField @dynamic_type {
   double get(int, double)
 }
 
+class org.elasticsearch.script.field.HalfFloatDocValuesField @dynamic_type {
+  float get(double)
```
Comment (stu-elastic) [inline modules/lang-painless/src/main/resources/org/elasticsearch/painless/org.elasticsearch.script.fields.txt:65]: Can you add a note explaining why this takes a double but returns a float.

**Review by stu-elastic — APPROVED** [review]
Very nice.

File: modules/lang-painless/src/main/resources/org/elasticsearch/painless/org.elasticsearch.script.fields.txt (hunk 1 above)
Comment (jdconrad) [reply modules/lang-painless/src/main/resources/org/elasticsearch/painless/org.elasticsearch.script.fields.txt:66]: Added.

**Comment by jdconrad** [comment]
@stu-elastic Thanks for the review. Will commit as soon as CI passes.

**Comment by jdconrad** [comment]
Due to the casting between floats and doubles, I updated the test to use close_to instead of match.

**Pull request merged**